install/kubernetes/cilium/charts/agent/templates/daemonset.yaml

```diff
@@ -157,12 +157,20 @@ spec:
           {{- toYaml .Values.resources | trim | nindent 10 }}
 {{- end }}
         name: cilium-agent
-{{- if .Values.global.prometheus.enabled }}
+{{- if or .Values.global.prometheus.enabled .Values.global.hubble.metricsServer }}
         ports:
+{{- if .Values.global.prometheus.enabled }}                
         - containerPort: {{ .Values.global.prometheus.port }}
           hostPort: {{ .Values.global.prometheus.port }}
           name: prometheus
           protocol: TCP
+{{- end }}
+{{- if .Values.global.hubble.metricsServer }}
+        - containerPort: {{ regexReplaceAll ":([0-9]+)$" .Values.global.hubble.metricsServer "${1}" }}
+          hostPort: {{ regexReplaceAll ":([0-9]+)$" .Values.global.hubble.metricsServer "${1}" }} 
```

Comment:
I'm not sure if this should be addressed here or in a follow up. For consistency across hubble and cilium, the user should specify the metrics server in the same way it configures it for Cilium. `--set global.hubble.metricsServer` is setting IP and port. In Cilium there's a flag dedicated for the address and another dedicated for port. cc @gandro

Reply:
I agree, but let's do that in a follow-up.

Reply:
So we will address this in the follow-up